src/devices/Ili9341/Ili9341.cs

```diff
@@ -0,0 +1,269 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.Drawing;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Ili9341
+{
+    /// <summary>
+    /// The ILI9341 is a QVGA (Quarter VGA) driver integrated circuit that is used to control 240×320 VGA LCD screens.
+    /// </summary>
+    public partial class Ili9341 : IDisposable
+    {
+        private const int ScreenWidthPx = 240;
+        private const int ScreenHeightPx = 320;
+        private const int DefaultSPIBufferSize = 0x1000;
+        private const byte LcdPortraitConfig = 8 | 0x40;
+        private const byte LcdLandscapeConfig = 44;
+
+        private readonly int _dcPinId;
+        private readonly int _resetPinId;
+        private readonly int _backlitPin;
+        private readonly int _spiBufferSize;
+        private readonly bool _shouldDispose;
+
+        private SpiDevice _spiDevice;
+        private GpioController _gpioDevice;
+
+        /// <summary>
+        /// Initializes new instance of ILI9341 device that will communicate using SPI bus.
+        /// </summary>
+        /// <param name="spiDevice">The SPI device used for communication. This Spi device will be displayed along with the ILI9341 device.</param>
+        /// <param name="dataCommandPin">The id of the GPIO pin used to control the DC line (data/command).</param>
+        /// <param name="resetPin">The id of the GPIO pin used to control the /RESET line (data/command).</param>
+        /// <param name="backlightPin">The pin for turning the backlight on and off, or -1 if not connected.</param>
+        /// <param name="spiBufferSize">The size of the SPI buffer. If data larger than the buffer is sent then it is split up into multiple transmissions. The default value is 4096.</param>
+        /// <param name="gpioController">The GPIO controller used for communication and controls the the <paramref name="resetPin"/> and the <paramref name="dataCommandPin"/>
+        /// If no Gpio controller is passed in then a default one will be created and disposed when ILI9341 device is disposed.</param>
+        /// <param name="shouldDispose">True to dispose the Gpio Controller</param>
+        public Ili9341(SpiDevice spiDevice, int dataCommandPin, int resetPin, int backlightPin = -1, int spiBufferSize = DefaultSPIBufferSize, GpioController? gpioController = null, bool shouldDispose = true)
```

Comment:
From what I can tell the ili9341 LCD controller is always 240x320. Do you have another resource to point to? If this will never be anything but that size, it seems like an unnecessary complication to add to this API

Reply:
I'm fine for a first merge like this. I have at least one external screen where the size of the screen is smaller than the driver controller. But ok, it's not much on the developer side, it's about adding an offset to x and y when writing and making sure it's not more than the size of the screen.
Again, the idea is to get all the small displays better over time with what we found. I understand you used the existing one as a base.

Reply:
I'll resolve this to make sure we don't miss any comment when merging and there doesn't seem to be any work missing in this PR